wrapper.go

```diff
@@ -0,0 +1,135 @@
+package sfxlambda
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-lambda-go/lambdacontext"
+	"github.com/signalfx/golib/datapoint"
+	"os"
+	"reflect"
+	"strings"
+	"time"
+)
+
+// HandlerFuncWrapper is the interface that wraps the lambda handler function.
+type HandlerFuncWrapper interface {
+	GetWrappedHandlerFunc() func(context.Context, interface{}) (interface{}, error)
+	SendDatapoint(context.Context, *datapoint.Datapoint)
+}
+
+type handlerFuncWrapper struct {
+	wrappedHandlerFunc func(context.Context, interface{}) (interface{}, error)
+	defaultDimensions  map[string]string
+	notColdStart       bool
+}
+
+// NewHandlerFuncWrapper is the HandlerFuncWrapper factory function.
+func NewHandlerFuncWrapper(handlerFunc interface{}) HandlerFuncWrapper {
+	hfw := handlerFuncWrapper{}
+	hfw.wrappedHandlerFunc = func(ctx context.Context, payload interface{}) (interface{}, error) {
```

Comment:
You should propagate this provided context to everything that takes a context.  There shouldn't be any use of `context.Background` in code that is called from this wrapper.